### ISSUE TITLE
Invalid Union index (17); union only has 2 types

### ISSUE THREAD

**Issue by iNikem** [FasterXML/jackson-dataformats-binary#164]
I believe this is a different issue from #123.

Please see the project attached. Running `./gradlew run` demonstrates the failure.

We have a binary avro file serialised by Jackson. We have a writer schema in a separate file, which describes the class which was serialised. Next we have a reader schema with one of the fields from that class removed. In my actual project we have removed some fields from the class and updated a schema to be able to deserialised older data while ignoring removed fields. Unfortunately removing some fields from reader schema fails deserialisation.

The correct behaviour, as far as I understand it, should be to ignore fields which are not present in the reader schema.

[avro.zip](https://github.com/FasterXML/jackson-dataformats-binary/files/3205991/avro.zip)

**Comment by cowtowncoder** [comment]
Sounds like a bug, thank you for reporting this. I hope to look into it relatively soon (there's bit of a backlog).

**Comment by cowtowncoder** [comment]
I can reproduce this problem, added failing unit test. I suspect this is due to a problem in updating state after "missing" `long` field is skipped (or, likely, is not and should be skipped).

**Comment by wincentzzz** [comment]
Hello, any updates for this issue?

**Comment by cowtowncoder** [comment]
@wincentzzz No. The usual practice is to add a note on issue if there are updates.

**Comment by cowtowncoder** [comment]
Ok: turns out this is/was test issue: test was not properly using Reader AND Writer Schemas, as required to support different schemas & schema evolution.

Will be checking properly working test next.
